thunder/core/functionalization.py

```diff
@@ -566,7 +566,7 @@ def _reshape_bsym_ctor(src: TensorProxy, dst: TensorProxy, trace: Trace) -> tupl
         if bsym in bsym_to_copy_bsyms:
             functionalized_bsyms.extend(bsym_to_copy_bsyms[bsym])
             copy_bsym = functionalized_bsyms[-1]
-            swap_map_for_return[variableify(copy_bsym.flat_proxy_args[0])] = copy_bsym.flat_proxy_args[1]
+            swap_map_for_return[variableify(copy_bsym.flat_proxy_args[0])] = copy_bsym.flat_proxy_outs[0]
```

Comment:
For example, consider a function
```py
def f(x):
  x += 1
  return x
```
On this line, the intermediate trace looks like
```py
t0 = ltorch.add(x, 1)
t1 = prims.copy_(t0, x)  # src=t0, dst=x
return {"output": t0, "flat_args": (t0,)}  # All occurrences of x became t0 due to functionalization
```
Before this PR, `swap_map_for_return` swapped `t0` with `x`, because the returned tensor should point to the same data segment in the memory as the argument `x`.

In the new code, `t0` now gets swapped with `t1`. This forces the topological sort algorithm to order the return statement after the copy. This is okay in terms of correctness, because the return value of `copy_` is exactly the same as the destination tensor.
https://github.com/Lightning-AI/lightning-thunder/blob/d70b10e44244317c462c6068a7b6a44656d5901d/thunder/executors/torchex.py#L2145-L2147